todo/src/reducers/index.js

```diff
@@ -0,0 +1,61 @@
+import { ADDTODO, TOGGLETODO, CLEAR } from '../actions';
+
+const initialState = {
+  todos: [{
+    task: 'Organize Garage',
+    id: 1528817077286,
+    completed: false
+  },
+  {
+    task: 'Bake Cookies',
+    id: 1528817084358,
+    completed: false
+  },
+  {
+    task: 'Create TodoList',
+    id: 1528817084395,
+    completed: false
+  },
+  {
+    task: 'Learn Redux',
+    id: 1528817084975,
+    completed: false
+  }]
+}
+
+export const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADDTODO:
+    const newTodo = {
+      task: action.payload,
+      id: Date.now(),
```

Comment:
Date.now() is by definition an impure function -- it returns something different every time it's invoked. We want to keep everything inside our reducers pure, so this newly created `newTodo` could be made within your action creator instead, and the whole newTodo passed to the reducer as the payload instead of just the text